### DISCUSSION TITLE
pgwatch3 as K8S operator

### DISCUSSION THREAD

**Discussion by ondar** [cybertec-postgresql/pgwatch#569] [Q&A] [answered]
- Has anyone already implemented K8S helm charts for pgwatch3?
- Is it actually possible to have postgres and prometheus sinks at the same time? With multiwriter?
- Any idea about resource usage for 1 source and 2 sinks (timescale postgres + prometheus)?
- It is possible to disable UI to reduce resource usage and increase security?

**Comment by pashagolub** [comment, marked as answer]
> * Has anyone already implemented K8S helm charts for pgwatch3?

There are some ongoing works. @Schmaetz any news on this?

 
> * Is it actually possible to have postgres and prometheus sinks at the same time? With multiwriter?

Yes, this is by design. Yes, multiwriter is responsible for that. Just put several `--sink=` command-line parameters and pgwatch will catch those automatically

> 
> * Any idea about resource usage for 1 source and 2 sinks (timescale postgres + prometheus)?
> 

Minimal. Depends on the intervals of metrics and scrapping. If compression is on for TimescaleDB and we assume
45M rows of real world stat_statements data (6db * 3mo), storage: 8 GB compressed.

But usually people have shorter retention periods, so it's even less.

>  * It is possible to disable UI to reduce resource usage and increase security?

Implemented in #576, just make sure `--web-addr=''` is empty

**Reply by ondar** [reply]
Thanks, I will update docs with this new info. And what about memory + cpu usage?

**Reply by pashagolub** [reply]
One needs to measure but I don't think it drastically increased comparing to previous version

**Reply by ondar** [reply]
Is it possible to somehow configure multiple sinks with PW_SINK env var, or you must use --sink cmd-line parameter? Same with --web-addr=''?

**Reply by pashagolub** [reply]
I didn't try to use several values with env var, need to check. if `PW_WEB_ADDR` is empty that means default value `:8080` will be used, I suppose. What is the problem using hard coded command-line parameter to disable web ui?

**Reply by ondar** [reply]
Probably no problem, depends how those helm charts are created. I'm no expert at that, I'm just relaying questions from other people from our company. I will have more answers and maybe questions probably tomorrow.

**Comment by JecminekTomas** [comment]
Hello, any chance of progression on pgwatch3 K8s helm charts? :)

**Reply by pashagolub** [reply]
I won't do it. Not my area of interest. And the work should be done in https://github.com/cybertec-postgresql/pgwatch-charts

**Reply by JecminekTomas** [reply]
Understand.

**Reply by JecminekTomas** [reply]
Open for PRs?

**Reply by pashagolub** [reply]
Sure! :) PRs are always welcome!

**Comment by Schmaetz** [comment]
@JecminekTomas https://github.com/cybertec-postgresql/pgwatch-charts/tree/pgwatch-3-helm-chart/helm/pgwatch

just didn't find the time to continue working. 
the next item on my agenda is the sink theme for prometheus.

**Reply by JecminekTomas** [reply]
@Schmaetz, completely understand, I might find some spare-time to continue on this, because it is pretty important for us. I've been searching through the whole internet and only viable solution is pgwatch, but we just need those helm charts.

**Reply by Schmaetz** [reply]
@JecminekTomas 
understand, just check the current status and let me know what is missing for you. At the moment its very basic. Its usable on k8s and openshift.

**Reply by ondar** [reply]
We do have some working charts, but I'm not the one responsible for them. Can we share something @MichaelZhasil?
